resources/manifests/psp-restricted.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: restricted
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
+    seccomp.security.alpha.kubernetes.io/defaultProfileName:  'docker/default'
+spec:
+  privileged: false
+  # Required to prevent escalations to root.
+  allowPrivilegeEscalation: false
+  # This is redundant with non-root + disallow privilege escalation,
+  # but we can provide it for defense in depth.
+  requiredDropCapabilities:
+  - KILL
+  - MKNOD
+  - SETUID
+  - SETGID
+  # Allow core volume types.
+  volumes:
+  - 'configMap'
+  - 'emptyDir'
+  - 'projected'
+  - 'secret'
+  - 'downwardAPI'
+  # Assume that persistentVolumes set up by the cluster admin are safe to use.
+  - 'persistentVolumeClaim'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  runAsUser:
+    # Require the container to run without root privileges.
+    rule: 'MustRunAs'
```

Comment:
Why not just `MustRunAsNonRoot`?

When testing with the current version, I was able to deploy ingress-nginx w/o modifications and wonder why (running as uid `33`). Ideas?

Reply:
When this is enabled, kubelet does not allow a pod to start that has non-numeric user, hence I have explicitly added UIDs

You would see an error like this on pod
```
Error: container has runAsNonRoot and image has non-numeric user (www-data), cannot verify user is non-root
```

Reply:
OK, good to know, thanks.